src/ui/Finished.tsx

```diff
@@ -1,23 +1,27 @@
 import { useState } from 'react'
 import { reset, useStore } from '../store'
-import { getScores, insertScore } from '../data'
+import { getScores, insertScore, ScoreInterface } from '../data'
 import { Score } from './LeaderBoard'
 import { Auth } from './Auth'
+import { Session } from '@supabase/supabase-js'
 
 export const Finished = () => {
   const LOCAL_STORAGE_KEY = 'racing-pmndrs-name'
 
-  const [finished, session] = useStore((state) => [state.finished, state.session])
+  const [finished, session] = useStore((state) => [state.finished, state.session]) as  [number, Session | null]
   const readableTime = (finished / 1000).toFixed(2)
 
-  const [name, setName] = useState(window.localStorage.getItem(LOCAL_STORAGE_KEY))
-  const [scores, setScores] = useState(null)
-  const [position, setPosition] = useState(null)
+  const [name, setName] = useState(window.localStorage.getItem(LOCAL_STORAGE_KEY)) as [string, React.Dispatch<React.SetStateAction<string>>]
+  const [scores, setScores] = useState<ScoreInterface[]>(null!)
+  const [position, setPosition] = useState<number>(null!)
 
   const sendScore = async () => {
-    window.localStorage.setItem(LOCAL_STORAGE_KEY, name)
-    const [{ id }] = await insertScore({ time: finished, name })
+    window.localStorage.setItem(LOCAL_STORAGE_KEY, name as string)
+    const [{ id }] = await insertScore({ time: finished as number, name }) as ScoreInterface[]
```

Comment:
I think we should make finished a number so we don't do have to do things like this.